erpnext/accounts/page/pos/pos.js

```diff
@@ -1779,6 +1779,13 @@ erpnext.pos.PointOfSale = erpnext.taxes_and_totals.extend({
 		})
 	},
 
+	validate_item_condition: function (data, item) {
+		var apply_on = frappe.model.scrub(data.apply_on);
+
+		return (data.apply_on == 'Item Group'
```

Comment:
IG 0
    - IG 01
        - IG 011
        - IG 012
    - IG 02

For the above Item group tree, if an item is belong to IG 012, the the pricing rule applicable to IG0 and IG01 should be evaluated.